src/app/dim-api/reducer.ts

```diff
@@ -199,6 +199,14 @@ export const dimApi = (
 
     case getType(actions.profileLoaded): {
       const { profileResponse, account } = action.payload;
+
+      // If the user was using "element" sort, replace it with "elementWeapon" and "elementArmor" sorts.
+      const sortOrder = profileResponse.settings?.itemSortOrderCustom || [];
+
+      if (sortOrder.includes('element')) {
+        sortOrder.splice(sortOrder.indexOf('element'), 1, 'elementWeapon', 'elementArmor');
+      }
```

Comment:
This should probably also migrate the `sortReversals` in the same manner -- DIM's sort options can sort items ascending or descending and the new options should probably retain that.

Reply:
@robojumper How should I handle `sortReversals`? 

For `sortOrder` I used `profileResponse.settings?.itemSortOrderCustom` to get the users sort order. There seems to be no `sortReversals` or `itemSortReversals` properties on type `Settings` to do the same for the reversed order. Is there a better method to getting the reversed sort settings? Or maybe I did the `sortOrder` part weird initially and that is messing with me?

Reply:
because this is using dim-api Profile type, and not DIM's adjusted local settings types, they need reconciliation so that reversals will appear here.
need to merge this first https://github.com/DestinyItemManager/dim-api/pull/105

Reply:
It seems that the pull request linked above was merged a few days ago. I am still having the error `Property 'itemSortReversals' does not exist on type 'Settings'.` but in my testing it seems to still work. Now if the user has the sort "Element" reversed it should be replaced with both a reversed "Armor Element" and "Damage Type" sort. Let me know if you know why that error is occurring and if there is anything else that needs changed.

Reply:
The updated package is published but not used by DIM yet. Run `yarn upgrade @destinyitemmanager/dim-api-types` and commit the lockfile, then the error should go away.